datasets/_global/global_energy_own/crawler.py

```diff
@@ -0,0 +1,141 @@
+import openpyxl
+from typing import Dict
+from pantomime.types import XLSX
+
+from zavod import Context
+from zavod import helpers as h
+
+IGNORE = [
+    "sequence_no",
+    "us_eia_860",
+    "parent_entity_ids",
+    "parents",
+    "headquarters_country",
+    "publicly_listed",
+    "decision_date",
+    "gcpt_announced_mw",
+    "gcpt_cancelled_mw",
+    "gcpt_construction_mw",
+    "gcpt_mothballed_mw",
+    "gcpt_operating_mw",
+    "gcpt_permitted_mw",
+    "gcpt_pre_permit_mw",
+    "gcpt_retired_mw",
+    "gcpt_shelved_mw",
+    "gogpt_announced_mw",
+    "gogpt_cancelled_mw",
+    "gogpt_construction_mw",
+    "gogpt_mothballed_mw",
+    "gogpt_operating_mw",
+    "gogpt_pre_construction_mw",
+    "gogpt_retired_mw",
+    "gogpt_shelved_mw",
+    "gbpt_announced_mw",
+    "gbpt_construction_mw",
+    "gbpt_mothballed_mw",
+    "gbpt_operating_mw",
+    "gbpt_pre_construction_mw",
+    "gbpt_retired_mw",
+    "gbpt_shelved_mw",
+    "gbpt_cancelled_mw",
+    "gcmt_proposed_mtpa",
+    "gcmt_operating_mtpa",
+    "gcmt_shelved_mtpa",
+    "gcmt_mothballed_mtpa",
+    "gcmt_cancelled_mtpa",
+    "gspt_operating_ttpa",
+    "gspt_announced_ttpa",
+    "gspt_construction_ttpa",
+    "gspt_retired_ttpa",
+    "gspt_operating_pre_retirement_ttpa",
+    "gspt_mothballed_ttpa",
+    "gspt_cancelled_ttpa",
+    "total",
+]
+
+
+def crawl_company(context: Context, row: Dict[str, str]):
+    id = row.pop("entity_id")
+    name = row.pop("entity_name")
+    if name == "unknown":
+        return
```

Comment:
if there are entities that you skip, you need a way to skip those relations too. One way to do that is to create a set `skipped` in crawl() which you pass into create_company which adds an ID there if it skips them, and pass into crawl_ownership which checks if both sides of the relationship are in there before emitting

```
 (global-energy-ownership-tracker) $ zavod validate --clear datasets/_global/global_energy_own/global_energy_own.yaml 2>&1 |head -15
2024-09-30 13:56:20 [info     ] Loading linker from: data/resolver.ijson [zavod.dedupe]
2024-09-30 13:56:21 [info     ] Loaded 484175 clusters from: data/resolver.ijson [nomenklatura.resolver.resolver]
2024-09-30 13:56:21 [info     ] Building local LevelDB aggregator... [zavod.store] scope=global_energy_own
2024-09-30 13:56:21 [info     ] Indexing aggregator...         [zavod.store] dataset=global_energy_own scope=global_energy_own statements=50000
2024-09-30 13:56:22 [info     ] Indexing aggregator...         [zavod.store] dataset=global_energy_own scope=global_energy_own statements=100000
2024-09-30 13:56:22 [info     ] Indexing aggregator...         [zavod.store] dataset=global_energy_own scope=global_energy_own statements=150000
2024-09-30 13:56:22 [info     ] Local LevelDB aggregator is ready. [zavod.store] scope=global_energy_own statements=153999
2024-09-30 13:56:22 [info     ] Validating dataset             [global_energy_own] dataset=datasets/global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-001aa73491f07151be929bbecc47ddd1a269c136 property asset references missing id energy-monitor-e100000124736 [global_energy_own] dataset=global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-00250bd040e6692bd6764031c191d7f743cc44db property owner references missing id energy-monitor-e100000132388 [global_energy_own] dataset=global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-00639e02516817be7c953c93e5850ca9cc1245d0 property asset references missing id energy-monitor-e100000124743 [global_energy_own] dataset=global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-00755b38a0f5b639c8841942155b6719f8f2e6b7 property asset references missing id energy-monitor-e100000128991 [global_energy_own] dataset=global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-0075f56c7fbf7198b5d397a643ff3b41dfedfe0e property owner references missing id energy-monitor-e100000123587 [global_energy_own] dataset=global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-007724cbf40f097ae7d356662f88d6f42f870a66 property asset references missing id energy-monitor-e100000124591 [global_energy_own] dataset=global_energy_own
2024-09-30 13:56:22 [warning  ] energy-monitor-0087080f1d132e0c9eb34f6e92975c5ca7270306 property asset references missing id energy-monitor-e100000125390 [global_energy_own] dataset=global_energy_own
```